scripts/parser-tests/typescript/index.js

```diff
@@ -5,9 +5,9 @@ import TestRunner from "../utils/parser-test-runner.js";
 import ErrorCodes from "./error-codes.js";
 
 const getEncoding = path =>
-  ({ fffe: "utf-16le", feff: "utf-16be" }[
+  ({ fffe: "utf-16le", feff: "utf-16be" })[
     fs.readFileSync(path).slice(0, 2).toString("hex")
-  ] || "utf-8");
+  ] || "utf-8";
```

Comment:
Would the old behavior be clearer?

Reply:
This changed because we changed

```diff
- ({ fffe: "utf-16le", feff: "utf-16be" }[
+ ({ fffe: "utf-16le", feff: "utf-16be" })[
  fs.readFileSync(path).slice(0, 2).toString("hex")
- ] || "utf-8");
+ ] || "utf-8";
```

in  2.8.2 https://github.com/prettier/prettier/blob/main/CHANGELOG.md#add-parentheses-to-head-of-expressionstatement-instead-of-the-whole-statement-14077-by-fisker


And we format [array member access similarly](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEBzOMCiUkAJgJZTaYC8mADgIYwAW1AfADpSaYAUwmAMyFwkmdiACuMAQFoAjADYANnHEAaQXCGjxU2YoBGqkJgC+AbU7duAtADoATnHrEAYqRUBlAJ5EeDZgBKOzQlUjA4HgAGDQAmYJgITxgHcmwecSY4AA9xQKtMAF1MAB8SsUlpGQAOPIBuTk5IKAwcPEISNOo6RhYqDi5Mc34hAREKvXllYw0x7Qmqw2MzQstBm3snF3cvXzB-XuDQ8MiYzHi7ROTUigyQLNyQfMHisoXZWpA6kDUQCFoYKR0MhQPQHA4IAB3AAKYIQaGQIHoSkh9G8CN+Bgc9DAAGs8J56ABbOAAGXIcGQAmRaDgmOxeIJDDAaWQKQkdJAcCJRmIxDgxFJ9AoEnouFcEAcRMYgIoiPoUggP3uMCJSgA6kxSPA0My4J54drSAA3bXeRFgNAYkDkWkOGDQ7HYaVUmmcgBWaGynjSKgAihIIPBXUpab8GA47YiDPQjEplbQbjB1aRiMxkNUYiBExBaersbREYm4HbjZTfgBHQPwR3-BEoehoGRQOACgXKpxV0hOR1il1Iamhzm0omkNkODm-NC+uAAQRgKVIBikcGhcAc5JbIbDIGnFH91cpA7dvxgsZTaaYyFip+xHjSAGEIET+1y0ABWZUSWkAFVj9cHHdjQ5ABJKB+VgTwwFSAFZ3A5JvBUbc4FMUwgA).

Reply:
Thanks for the explanation!
I agree that the example in https://github.com/prettier/prettier/issues/13999#issuecomment-1354050144 would look better.
```js
// Before:
(function () {} ());

// After:
(function () {})();
```
But here I am more inclined to the original behavior, because the original format I can see at a glance that `"utf-8"` is the return value of this arrow function, and the new format looks easy to match `arrowFn||"utf -8"` Confused.
Of course this is not very important and will not block this PR.